doc/PLUGINS.md

```diff
@@ -1001,7 +1001,25 @@ onion fields which a plugin doesn't want lightningd to consider.
 ```
 
 `fail` will tell `lightningd` to fail the HTLC with a given hex-encoded
-`failure_message` (please refer to the [spec][bolt4-failure-messages] for details: `incorrect_or_unknown_payment_details` is the most common).
+`failure_message` (please refer to the [spec][bolt4-failure-messages] for
+details: `incorrect_or_unknown_payment_details` is the most common).
+
+
+```json
+{
+  "result": "fail",
+  "failure_onion": "[292bytes of a serialized error packet]"
```

Comment:
The error onion has a variable size, so it'd be better not allude to a fixed size here by putting the `292bytes` here.

Reply:
Does it? I was pretty sure it had a fixed size.

This proves I shouldn't be writing these things.

Reply:
You are for sure not more or less talented than I am :)
It's about learning not knowing

Reply:
Why has it a variable size? The spec says it is 260 bytes plus an HMAC of 32 bytes, then it is XORed against a cipherstream of the same size on every hop. I don't understand how it can be different than 292 bytes.

Reply:
Here: https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#returning-errors

It says the packet consists of `hmac(32) + failure_len(2) + failure + pad_len(2) + pad` such that `failure + pad = 256`. The `channel_update` and other things I imagine are contained in the `failure` element above.